tests/unit/channel_test.py

```diff
@@ -68,3 +74,88 @@ def test_read_bulk_response(ssh_client_session):
             break
 
     assert b'Hello World' in response  # noqa: WPS302
+
+
+def test_request_exec(ssh_channel):
+    """Test direct call to request_exec."""
+    ssh_channel.request_exec('exit 1')
+
+    rc = -1
+    while rc == -1:
+        ssh_channel.poll(timeout=POLL_EXIT_CODE_TIMEOUT)
+        rc = ssh_channel.get_channel_exit_status()
+    assert rc == 1
+
+
+def test_send_eof(ssh_channel):
+    """Test send_eof correctly terminates input stream."""
+    ssh_channel.request_exec('cat')
+    ssh_channel.send_eof()
+
+    rc = -1
+    while rc == -1:
+        ssh_channel.poll(timeout=POLL_EXIT_CODE_TIMEOUT)
+        rc = ssh_channel.get_channel_exit_status()
+    assert rc == 0
+
+
+def test_send_signal(ssh_channel):
```

Comment:
@pbrezina any idea why this test might be SEGFAULTing sometimes? See the CI log: https://github.com/ansible/pylibssh/actions/runs/7798061967/job/21271285995?pr=576#step:15:319.

Reply:
If I read it correctly this test timed out when connecting to ssh server:
```
 >       raise TimeoutError('Timed out waiting for a successful connection')
E       TimeoutError: Timed out waiting for a successful connection
```

I saw this from time to time when running tests locally (also with the tests not related to my changes).

Segfault comes from
```
XFAIL tests/unit/channel_test.py::test_exec_command - reason: This test causes SEGFAULT, flakily. Ref: https://github.com/ansible/pylibssh/issues/57; pytest-forked reason: :-1: running the test CRASHED with signal 11
```

Reply:
Which also means that #57 is not fixed by this.

Reply:
```
        libssh.ssh_channel_send_eof(channel)
        result.returncode = libssh.ssh_channel_get_exit_status(channel)
        if channel is not NULL:
            libssh.ssh_channel_close(channel)
            libssh.ssh_channel_free(channel)
```
I think you need to poll for exit status since it may still return -1 and you close the channel. Perhaps there is some race condition when the channel is being closed but callback is still called. We could be sure if we have coredump available but this is my wild guess.

Reply:
Would you be able to take a look? I kinda don't want to make the CI even more flaky with this patch..

Reply:
It's not getting more flaky since the problems are already there :-)

Anyway, I believe I have fixed the timeout issue. See the last commit.

For the segfault... is it possible to get a coredump and full backtrace out of the tests? I never hit it locally and I don't think we can fix it without it. We can try to implement my proposal, but then again, if it was true, it should be possible to write a solid reproducer and I was not able to find one.

Reply:
Well, I googled and couldn't understand how to retrieve that core. The faulthandler is enabled and prints out some trace through `catchsegv`. I wish there was a way to put it into a GHA artifact, though.

Thanks for the patch!